File: Practice/Easy/1089.DuplicateZeros/python/solution2.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        copy = arr.copy()
        counter = 0
        for i in range(len(copy)):
            if copy[i] == 0:
                arr.insert(i + counter, 0)
                counter += 1
                arr.pop()

File: Practice/Easy/1089.DuplicateZeros/js/solution2.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function(arr) {
    let copy = [...arr];
    let counter = 0;

    for (let i = 0; i < copy.length; i++) {
        if (copy[i] == 0) {
            arr.splice(i + counter, 0, 0);
            counter++;
            arr.pop();
        }
    }

};